BenadrylAutomation/src/main/java/com/jnj/pageobjects/AllergiesPage.java

```diff
@@ -0,0 +1,128 @@
+
+	private WebElement getRelatedArticleCard(String index) {
+		return driver.findElement(By.xpath("(//div[@class='taco-inner'])['" + index + "']"));
+	}
+
+	/**
+	 * Function to verify url and banner title
+	 * 
+	 * @param expectedUrl   pass expected URL
+	 * @param expectedTitle pass expected title
+	 */
+	public void verifyUrlAndTitle(String expectedUrl, String expectedTitle) {
+		Action.verifyPageUrl(expectedUrl);
+		String actualTitle = driver.getTitle();
+		Action.printAndAssert(actualTitle, expectedTitle);
+	}
+
+	/**
+	 * Verifies if the banner is displayed and checks if the heading matches the
+	 * expected heading.
+	 * 
+	 * @param expectedHeading The expected heading text of the banner.
+	 */
+	public void verifyBanner(String expectedHeading) {
+		extentInfoLog("Banner is displayed : ", Action.isDisplayed(driver, bannerImg));
+		String actualHeading = bannerHeading.getText();
+		Assert.assertEquals(actualHeading, expectedHeading);
+		extentInfoLog("Heading is displayed : ", actualHeading);
+	}
+
+	/**
+	 * Verifies if the header banner is displayed, clicks on the specified header
+	 * element, and checks if the content scrolls to the specified heading.
+	 * 
+	 * @param name    The name of the header element to click on.
+	 * @param heading The heading text to check if content scrolls to.
+	 */
+	public void verifyHeaderScrolling(String name, String heading) {
+		extentInfoLog("Heading banner is displayed : ", Action.isDisplayed(driver, headerBanner));
+		Action.performActionwithExtentInfoLog(this.getAllergiesHeader(name), "click",
+				"Clicking on : " + getAllergiesHeader(name).getText());
+		extentInfoLog("Content Scrolled to : ", Action.isDisplayed(driver, getAllergiesScroll(heading)));
+	}
+
+	/**
+	 * Verifies if the header banner is displayed, clicks on the specified header
+	 * element, and checks if the content scrolls to the specified heading.
+	 * 
+	 * @param name    The name of the header element to click on.
+	 * @param heading The heading text to check if content scrolls to.
+	 */
+	public void verifySecondHeaderScrolling(String name, String headingScroll) {
+		extentInfoLog("Heading banner is displayed : ", Action.isDisplayed(driver, headerBanner));
+		Action.performActionwithExtentInfoLog(this.getAllergiesHeader(name), "click",
+				"Clicking on : " + getAllergiesHeader(name).getText());
+		extentInfoLog("Content Scrolled to : ", Action.isDisplayed(driver, getAllergyScroll(headingScroll)));
+	}
+
+	/**
+	 * Verifies if the related content card is displayed for the specified index.
+	 * 
+	 * @param index The index of the related content card to verify.
+	 */
+	public void verifyRelatedContent(String index) {
+		extentInfoLog("Content card is displayed : ", Action.isDisplayed(driver, getRelatedArticleCard(index)));
+	}
+
+	/**
+	 * Verifies if the related product is displayed.
+	 */
+	public void verifyRelatedProduct() {
+		extentInfoLog("Related Product is displayed : ", Action.isDisplayed(driver, relatedProduct));
```

Comment:
Where is the assertion @RashiTiwari21 ?

Reply:
@goutamnaik007 removed this function and in the test file calling relatedProduct() from compare page.

Reply:
Verifed @vaibhavNagvekarSJI